### ISSUE TITLE
Enable typings of extended MUI props in Storybook

### ISSUE THREAD

**Issue by Ajalle** [clevercanary/data-explorer#106]
### Our current flow (with example of Alert component)

We have overrides of MUI props in types/data-explorer-ui.d.ts :
```
declare module "@mui/material/Alert" {
  interface AlertPropsVariantOverrides {
    banner: true; // Single line alert.
    neutral: true;
  }
}
```

We are creating new variants of Alert by modifying MUI theme: 
```
export const MuiAlert = (theme: Theme): Components["MuiAlert"] => {
  return {
    variants: [
      {
        props: { severity: "info", variant: "neutral" },
        style: {
          backgroundColor: theme.palette.smoke.light,
          padding: 16,
        },
      },
      {
        props: { variant: "banner" },
        style: {
          padding: 16,
          // eslint-disable-next-line sort-keys -- disabling key order for readability
          "& .MuiAlert-icon": {
            padding: 0,
          },
          "& .MuiAlertTitle-root": {
            ...theme.typography[TEXT_BODY_500],
          },
        },
      },
 ]}
```

### The problem

When we want to create component that uses MUI Alert, those overrides are being ignored by Storybook
```
import {
  Alert as MAlert,
  AlertProps as MAlertProps,
} from "@mui/material";

export interface AlertProps {
  variant?: MAlertProps["variant"];
}
```
Storybook would render only `"filled" | "outlined" | "standard"` while actual type we have is `"banner" | "filled" | "outlined" | "standard" | "neutral"`

We should investigate why it happens and if there is a way to fix it in once for overrides of all MUI components.

**Comment by Ajalle** [comment]
I couldn't find any solution. Things I tried:
1. Convert config to .ts
2. Add typescript configuration to config: https://storybook.js.org/docs/react/configure/typescript
3. Play with ts config: https://github.com/storybookjs/storybook/issues/8961

The issue seems to be related to underlaying `"react-docgen-typescript"` parser (https://github.com/styleguidist/react-docgen-typescript), which transforms typescript definitions to arguments. Storybook gives very limited configuration to plugin, I tried multiple options.
The parser itself allows to set custom typescript config into it, but there is a low chance it would work and I couldn't find how to do it.

If someone else wants to continue with investigation, you are welcome.

For now, I'd say that we should simply add custom properties manually, there is no much work to do it. I am adding extra properties for Alert as new PR and link it to this issue.
